### PR TITLE
Add nonces to script tags

### PR DESCRIPTION
**Pull request by labkey-adam**
#### Rationale
CSPs like nonces. Also, path-first URL are preferred.

### CODE REVIEW

File: src/org/labkey/test/tests/FileAttachmentColumnTest.java [hunk 1]
```diff
@@ -100,7 +100,7 @@ private void doSetup()
 
     private void createList()
     {
-        beginAt("/project/" + getProjectName() +"/"+ FOLDER_NAME + "/begin.view?");
+        beginAt(getProjectName() + "/" + FOLDER_NAME + "/begin.view");
```
Comment (labkey-tchad) [inline src/org/labkey/test/tests/FileAttachmentColumnTest.java]: Missed the controller.
```suggestion
        beginAt(getProjectName() + "/" + FOLDER_NAME + "/project-begin.view");
```

File: src/org/labkey/test/tests/AdminConsoleTest.java [hunk 2]
```diff
@@ -68,7 +68,7 @@ public void testRibbonBar()
         waitForElement(Locator.xpath("//div[contains(text(), 'Cannot enable the ribbon message without providing a message to show')]"));
 
         String linkText = "and also click this...";
-        String html = "READ ME!!!  <a href='<%=contextPath%>/project/home/begin.view'>" + linkText + "</a>";
+        String html = "READ ME!!!  <a href='<%=contextPath%>" + "/home/project-begin.view'>" + linkText + "</a>";
```
Comment (labkey-tchad) [inline src/org/labkey/test/tests/AdminConsoleTest.java:71]: Looks like some structural search weirdness
```suggestion
        String html = "READ ME!!!  <a href='<%=contextPath%>/home/project-begin.view'>" + linkText + "</a>";
```

**Review by labkey-tchad — CHANGES_REQUESTED** [review]
One bad URL change.
Everything in `data` is used to populate wikis; I thought wikis didn't need to have the nonces explicitly defined?

File: src/org/labkey/test/tests/AdminConsoleTest.java (hunk 2 above)
Comment (labkey-adam) [reply src/org/labkey/test/tests/AdminConsoleTest.java:71]: Looks a little funny in this case, but it is intentional. I use `<%=contextPath%>/` to search for controller-first URLs. Path-first URLs usually have `<%=contextPath%><%=containerPath%>`, but obviously not the case of a hard-coded container path like this. Separating like this means the URL won't show up in searches.

**Comment by labkey-adam** [comment]
> One bad URL change. Everything in `data` is used to populate wikis; I thought wikis didn't need to have the nonces explicitly defined?

You're correct... I'm not sure how I misread their usages. I think they're all corrected now.

**Review by labkey-tchad — APPROVED** [review]

**Pull request merged**